package/network/utils/comgt/files/ncm.sh

```diff
@@ -117,21 +127,23 @@ proto_ncm_setup() {
 	}
 
 	echo "Starting network $interface"
-	json_get_vars connect
-	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
-		echo "Failed to connect"
-		proto_notify_error "$interface" CONNECT_FAILED
-		return 1
-	}
-
 	echo "Setting up $ifname"
-	
+
 	proto_init_update "$ifname" 1
 	proto_add_data
 	json_add_string "manufacturer" "$manufacturer"
 	proto_close_data
 	proto_send_update "$interface"
 
+	json_load "$(cat /etc/gcom/ncm.json)"
+	json_select "$manufacturer"
+	json_get_vars connect
+	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
+		echo "Failed to connect"
+		proto_notify_error "$interface" CONNECT_FAILED
+		return 1
+	}
+
```

Comment:
Hi Matti,
thanks for your tips.
I try to summarize your proposal. The sequence to be issued is the following:
```
initialize:
AT+CFUN=1
AT+CGDCONT=1,"IP","web.vodafone.de",0,0
```
```
setpin:
AT+CPIN="****"
```
```
connect:
AT+CGACT=1,1
```

```
usb IF set up:
proto_init_update "$ifname" 1
```

```
finalize:
AT+CGDATA="M-MBIM",1,1
```

can you please confirm my understanding?